core/v2/serialization.go

```diff
@@ -0,0 +1,265 @@
+package v2
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"golang.org/x/crypto/sha3"
+)
+
+type abiG1Commit struct {
+	X *big.Int
+	Y *big.Int
+}
+type abiG2Commit struct {
+	X [2]*big.Int
+	Y [2]*big.Int
+}
+type abiBlobCommitments struct {
+	Commitment       abiG1Commit
+	LengthCommitment abiG2Commit
+	LengthProof      abiG2Commit
+	Length           uint32
+}
+type abiBlobHeader struct {
+	BlobVersion         uint8
+	BlobCommitments     abiBlobCommitments
+	QuorumNumbers       []byte
+	PaymentMetadataHash [32]byte
+}
+
+func blobHeaderArgMarshaling() []abi.ArgumentMarshaling {
+	return []abi.ArgumentMarshaling{
+		{
+			Name: "blobVersion",
+			Type: "uint8",
+		},
+		{
+			Name: "blobCommitments",
+			Type: "tuple",
+			Components: []abi.ArgumentMarshaling{
+				{
+					Name: "commitment",
+					Type: "tuple",
+					Components: []abi.ArgumentMarshaling{
+						{
+							Name: "X",
+							Type: "uint256",
+						},
+						{
+							Name: "Y",
+							Type: "uint256",
+						},
+					},
+				},
+				{
+					Name: "lengthCommitment",
+					Type: "tuple",
+					Components: []abi.ArgumentMarshaling{
+						{
+							Name: "X",
+							Type: "uint256[2]",
+						},
+						{
+							Name: "Y",
+							Type: "uint256[2]",
+						},
+					},
+				},
+				{
+					Name: "lengthProof",
+					Type: "tuple",
+					Components: []abi.ArgumentMarshaling{
+						{
+							Name: "X",
+							Type: "uint256[2]",
+						},
+						{
+							Name: "Y",
+							Type: "uint256[2]",
+						},
+					},
+				},
+				{
+					Name: "length",
+					Type: "uint32",
+				},
+			},
+		},
+		{
+			Name: "quorumNumbers",
+			Type: "bytes",
+		},
+		{
+			Name: "paymentMetadataHash",
+			Type: "bytes32",
+		},
+	}
+}
+
+func (b *BlobHeader) toABIStruct() (abiBlobHeader, error) {
+	paymentHash, err := b.PaymentMetadata.Hash()
+	if err != nil {
+		return abiBlobHeader{}, err
+	}
+	return abiBlobHeader{
+		BlobVersion: uint8(b.BlobVersion),
+		BlobCommitments: abiBlobCommitments{
+			Commitment: abiG1Commit{
+				X: b.BlobCommitments.Commitment.X.BigInt(new(big.Int)),
+				Y: b.BlobCommitments.Commitment.Y.BigInt(new(big.Int)),
+			},
+			LengthCommitment: abiG2Commit{
+				X: [2]*big.Int{
+					b.BlobCommitments.LengthCommitment.X.A0.BigInt(new(big.Int)),
+					b.BlobCommitments.LengthCommitment.X.A1.BigInt(new(big.Int)),
+				},
+				Y: [2]*big.Int{
+					b.BlobCommitments.LengthCommitment.Y.A0.BigInt(new(big.Int)),
+					b.BlobCommitments.LengthCommitment.Y.A1.BigInt(new(big.Int)),
+				},
```

Comment:
Should this in thr form of A1, A0 ?

Reply:
Each extension in G2Point is represented as `uint256[2]` [onchain](https://github.com/Layr-Labs/eigenlayer-middleware/blob/mainnet/src/libraries/BN254.sol#L44)